test/unit/services/instanceAdapter_spec.js

```diff
@@ -49,33 +67,181 @@ describe("The InstanceAdapter class ", function () {
-		var result;
-		var mapperStub;
+	describe("Getting an instance", function () {
+		describe("with a valid instanceId", function () {
+			var VALID_INSTANCE_ID;
+			var DEFAULT_TYPE = "t2.micro";
+			var DEFAULT_AMI  = "ami-d05e75b8";
+			var ID_REGEX     = /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/;
+			var instances    = new InstanceAdapter();
+
+			before(function () {
+				instances.createInstance()
+				.then(function (response) {
+					expect(response).to.be.an.instanceOf(Instance);
+					expect(response.id).to.match(ID_REGEX);
+					expect(response.type).to.equal(DEFAULT_TYPE);
+					expect(response.ami).to.equal(DEFAULT_AMI);
+					VALID_INSTANCE_ID = response.id;
+				});
+			});
+
+			it("gets a valid instance", function () {
+
+				instances.getInstance({ id : VALID_INSTANCE_ID })
+				.then(function (result) {
+					expect(result).to.be.an.instanceOf(Instance);
+					expect(result.id).to.match(ID_REGEX);
+					expect(result.type).to.equal(DEFAULT_TYPE);
+					expect(result.ami).to.equal(DEFAULT_AMI);
+				});
+			});
+		});
+
+		describe("with an invalid instanceId", function () {
+			var instances = new InstanceAdapter();
+
+			it("fails", function () {
+
+				instances.getInstance({ id : "foo-bar-baz" })
+				.then(function (result) {
+					expect(result).to.be.null;
+				});
+			});
+		});
+	});
+
+	describe("Querying for instances", function () {
+		var VALID_INSTANCE_ID;
 		var DEFAULT_TYPE = "t2.micro";
 		var DEFAULT_AMI  = "ami-d05e75b8";
+		var ID_REGEX     = /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/;
+		var instances    = new InstanceAdapter();
 
 		before(function () {
-			var mapper = new Genesis.MemoryMapper();
-			var instances = new InstanceAdapter(mapper);
 
-			mapperStub = Sinon.stub(mapper, "create")
-				.rejects(new Error("Simulated Failure"));
+			instances.createInstance()
+			.then(function (response) {
+				expect(response).to.be.an.instanceOf(Instance);
+				expect(response.id).to.match(ID_REGEX);
+				expect(response.type).to.equal(DEFAULT_TYPE);
+				expect(response.ami).to.equal(DEFAULT_AMI);
+				VALID_INSTANCE_ID = response.id;
+			});
+		});
 
-			return instances.createInstance(DEFAULT_AMI, DEFAULT_TYPE)
-			.catch(function (err) {
-				result = err;
+		describe("with valid parameters", function () {
+			it("receives an array of instances", function () {
+				instances.getAllInstances({ ami : DEFAULT_AMI, type : DEFAULT_TYPE })
+				.then(function (response) {
+					expect(response).to.be.an.instanceOf(Array);
+					expect(response.length).to.be.at.least(1);
+				});
 			});
 		});
 
-		after(function () {
-			mapperStub.restore();
+		describe("with no parameters", function () {
+			it("receives an array of all instances", function () {
+				instances.getAllInstances()
+				.then(function (response) {
+					expect(response).to.be.an.instanceOf(Array);
+					expect(response.length).to.be.at.least(1);
```

Comment:
This test is functionally no different that the previous one. How do we know that anything different happened? In other words, this test case doesn't actually prove that the query returns all of the instances and not just some arbitrary first instance.